collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -2674,6 +2675,16 @@ public void onCancelFormLoading() {
         finish();
     }
 
+    @Override
+    public void deleteOriginalFile(String questionIndex, String fileName) {
```

Comment:
I think we're getting close with the structure but I don't like that the `FormEntryActivity` implements the `QuestionMediaManager`. I think instead:

* the `AudioWidget` should take a `QuestionMediaManager` in as a dependency at it's constructor rather than casting the `context` to one.
* the `WidgetFactory` should also take a `QuestionMediaManager` in as one of the args to `createWidgetFromPrompt` so it can pass it to any widget that needs it. 
* `FormSaveViewModel#getMediaManager` should return a `QuestionMediaManager` not a `MediaManager`
* `FormSaveViewModel#getMediaManager` should be passed as the `QuestionMediaManager` argument to `WidgetFactory#createWidgetFromPrompt`
* `QuitFormDialogFragment`  should call `viewModel.getMediaManager().revertChanges` rather than relying on the static singleton instance

I'm not sure if I like that currently the `QuestionMediaManager` is simply owned by the `FromSaveViewModel` and then exposes it. I'm thinking that potentially the `FormSaveViewModel` could implement the `QuestionMediaManager` interface and then there would be no need for the `MediaManager` class. In that world the "revert" call (in `QuiteFormDialogFragment` could just be part of the logic `FormSaveViewModel#removeTempInstance`.

Reply:
Nice idea @seadowg, and Thanks for explaining so well. I implemented all the functionality of `MediaManager` in `FormSaveViewModel`, as you said if we have the `FormSaveViewModel` implements `QuestionMediaManager`, then `MediaManager` class is no longer required.